### PR TITLE
feat/orphan cache v0.20.0

### PR DESCRIPTION
**Pull request by lukeweb3**

### CODE REVIEW

File: nodedb.go [hunk 1]
```diff
@@ -104,6 +119,9 @@ func newNodeDB(db dbm.DB, cacheSize int, opts *Options) *nodeDB {
 		fastNodeCache:  cache.New(fastNodeCacheSize),
 		versionReaders: make(map[int64]uint32, 8),
 		storageVersion: string(storeVersion),
+
+		cachedOrphans: sync.Map{},
```
Comment (dumbeng) [inline nodedb.go]: This line can be removed

File: nodedb.go [hunk 2]
```diff
@@ -532,46 +551,125 @@ func (ndb *nodeDB) DeleteVersionsRange(fromVersion, toVersion int64) error {
...
 	// If the predecessor is earlier than the beginning of the lifetime, we can delete the orphan.
 	// Otherwise, we shorten its lifetime, by moving its endpoint to the predecessor version.
 	for version := fromVersion; version < toVersion; version++ {
-		err := ndb.traverseOrphansVersion(version, func(key, hash []byte) error {
+		orphans, ok := ndb.cachedOrphans.Load(version)
+		if !ok {
+			if err := ndb.loadOrphans(version); err != nil {
+				return err
+			}
+			orphans, ok = ndb.cachedOrphans.Load(version)
```
Comment (dumbeng) [inline nodedb.go:574]: Why trying to load from cache for the second time?

Comment (lukeweb3) [reply nodedb.go:574]: The preceding line, ndb.loadOrphans(version), stores the retrieved orphan nodes in cachedOrphans, so reload them to orphans. Here 'cachedOrphans' serves as a temporary variable.

File: nodedb.go (hunk 1 above)
Comment (lukeweb3) [reply nodedb.go]: removed.

**Pull request merged**